File: examples/threads/ThreadRunnableExample.java

/**
 * Shows how to create threads by implementing Runnable interface.
 */
class RunnableElement implements Runnable
{
    private String name;


    public RunnableElement(String s) {
	name = s;
    }


    public String getName() {
	return name;
    }


    public void run() {
	for (int i = 0; i < 10; i++) {
	    System.out.println("This is the " + this.getName() + " thread.");
	}
    }
}

public class ThreadRunnableExample
{
    public static void main(String args[]) {
	RunnableElement g1 = new RunnableElement("water");
	RunnableElement g2 = new RunnableElement("fire");
	RunnableElement g3 = new RunnableElement("earth");
	RunnableElement g4 = new RunnableElement("air");
	RunnableElement g5 = new RunnableElement("void");

	new Thread(g1).start();
	new Thread(g2).start();
	new Thread(g3).start();
	new Thread(g4).start();
	new Thread(g5).start();
    }
}

File: examples/golang/threads/ThreadRunnableExample.go
//Translation of the Java ThreadRunnableExample
//Author: Amit Jain & Paul Ellis
package main

import (
	"fmt"
	"sync"
)

//An Element stores a name
type Element struct {
	name string
}
//Create a new Element
func newElement(name string) *Element {
	e := Element{name: name}
	return &e
}
//Run and print out the Thread
func run(e *Element, wg * sync.WaitGroup) {

	for i := 0; i < 10000; i++ {
		fmt.Println("This is the " + e.name + " thread.")
	}
	wg.Done()
}

func main() {

	var wg sync.WaitGroup

	wg.Add(5)

	//Start the run function in a new "thread" called a goroutine
	go run(newElement("water"), &wg)
	go run(newElement("fire"), &wg)
	go run(newElement("earth"), &wg)
	go run(newElement("air"), &wg)
	go run(newElement("void"), &wg)

	wg.Wait()
}